subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GradleResolveVisitor.java

```diff
@@ -531,7 +532,11 @@ private boolean resolveFromDefaultImports(ClassNode type, boolean testDefaultImp
                 }
             }
             for (String gradlePublicPackage : gradlePublicPackages) {
-                if (resolveFromResolver(type, gradlePublicPackage + "." + name)) {
+                String fullName = gradlePublicPackage + "." + name;
+                if (resolveFromResolver(type, fullName)) {
+                    if ("org.gradle.util".equals(gradlePublicPackage)) {
+                        DeprecationLogger.nagUserWith("Default importing from org.gradle.util has been deprecated. You should import class " + fullName + " explicitly");
```

Comment:
:x:

I would make it explicit that by using a class in `org.gradle.util`, the build author is making use of an internal class. Something like:

> You are using class XXX from the `org.gradle.util` internal package. Automatic imports from this package are now deprecated. Either remove usage of this internal class (recommended), or add an explicit import on it (not recommended).

WDYT?

One problem with this is that you would have a different warning for each class. So maybe a better version is to _collect all classes_ used from `org.gradle.util`, and issue _one_ warning at the end of the compilation. @oehme WDYT?

Reply:
For so long I always see `org.gradle.util` as a public package ... I didn't realize it's internal. That's a good idea to collect all classes and output one warning. Let me see if it's feasible.

Reply:
I'd output one message per class and not try to aggregate them (which just makes this more complicated). Keep in mind that this is something that the user can fix very easily, so they will not see the messages for long.